.github/workflows/pre-commit.yaml

```diff
@@ -1,14 +1,51 @@
 name: pre-commit
-
 on:
-  pull_request:
   push:
-    branches: [main]
+    branches-ignore:
+      - 'main'
 
 jobs:
   pre-commit:
     runs-on: ubuntu-latest
+    if: >-
+      !contains(github.event.head_commit.message, 'ci skip') &&
+      (
+        startsWith(github.ref, 'refs/heads') ||
+        github.event.pull_request.draft == false
+      )
     steps:
-    - uses: actions/checkout@v3
-    - uses: actions/setup-python@v3
-    - uses: ./.github/actions/pre-commit
+      - name: Cancel Previous Runs
```

Comment:
can you not do this with concurrency settings now and avoid the third-party action?

Reply:
I imagine I could -- this is the default that `precommit::use_precommit(ci = "gha")` gives you. If you have a nice alternative, I'm happy to switch over.

Reply:
https://github.com/epinowcast/epinowcast/blob/90f7a5a05892a9d176cc71422b796b1c54aa24d6/.github/workflows/render-readme.yaml#L3